src/Concerns/Exportable.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace PodPoint\MailExport\Concerns;
+
+use Illuminate\Contracts\Mail\Mailer as MailerContract;
+use Illuminate\Mail\Mailable;
+use PodPoint\MailExport\Contracts\ShouldExport;
+
+/**
+ * @mixin Mailable
+ */
+trait Exportable
+{
+    /**
+     * Send the message using the given mailer.
+     *
+     * @param  \Illuminate\Contracts\Mail\Mailer  $mailer
+     * @return void
+     */
+    public function send(MailerContract $mailer)
+    {
+        $this->withSwiftMessage(function ($message) {
+            $message->_shouldStore = $this instanceof ShouldExport;
+            $message->_storageDisk = $this->storageDisk();
+            $message->_storagePath = $this->storagePath();
+            $message->_storageFilename = $this->storageFilename();
+        });
+
+        parent::send($mailer);
```

Comment:
Yeah as long as it extends the Laravel `Mailable` (which it should if it's a Mailable) it will work.

I didn't want to check that the trait was used on a `Mailable` properly because it's down to the developer to know what he's doing. Laravel `SerializesModel` trait for example doesn't check it's applied on a `Job` class.

Reply:
Okay I've got a question, what do you prefer between:

1. overriding the `send()` from our trait so that means you can just use the trait and implements the interface.
2. use the trait and add a separate `export()` to be called within the `build()` of each Mailable.

Option 1 main pro is that it's dead simple and easy to implement, matches the way Laravel does it with Queueable and ShouldQueue. Main con is that the `send()` function signature changes over time in the code, making it a bit more faffy and require to have "two flavours" of the package, one for L7+ and one for L5+6.

Option 2 main pro is that we don't need to have two flavours but instead you have to call our `export()` function manually.

Reply:
Jynx! I would say option 1 being most Laravel appeals more - I think developer experience trumps maintainer experience.. 😁